packages/preview-middleware-client/test/unit/flp/init.test.ts

```diff
@@ -298,5 +320,56 @@ describe('flp/init', () => {
 
             expect(sapMock.ui.require).toBeCalledWith([customInit]);
         });
+
+        test('init handle higher layer changes', async () => {
+            const flexSettings = {
+                layer: 'VENDOR',
+                pluginScript: 'my/script'
+            };
+            const reloadSpy = jest.fn();
+            const location = window.location;
+            Object.defineProperty(window, 'location', {
+                value: {
+                    reload: reloadSpy
+                }
+            });
+            VersionInfo.load.mockResolvedValue({ name: 'sap.ui.core', version: '1.84.50' });
+
+            // Mocking `sap.ui.require` to throw the correct error structure
+            sapMock.ui.require.mockImplementationOnce((libs, callback) => {
+                callback(
+                    async () => { throw 'Reload triggered'; },
+                    {}
+                );
+            });
+
+            const sendActionSpy = jest.spyOn(CommunicationService, 'sendAction').mockImplementation(() => { });
+            await init({ flex: JSON.stringify(flexSettings) });
+            const rendererCb = sapMock.ushell.Container.attachRendererCreatedEvent.mock.calls[0][0] as () => Promise<void>;
+
+            const mockService = { attachAppLoaded: jest.fn() };
+            sapMock.ushell.Container.getServiceAsync.mockResolvedValueOnce(mockService);
+
+            await rendererCb();
+
+            const loadedCb = mockService.attachAppLoaded.mock.calls[0][0] as (event: unknown) => Promise<void>;
+            await loadedCb({ getParameter: () => { } });
+
+            setTimeout(() => {
+                expect(sendActionSpy).toHaveBeenCalled();
+                expect(sendActionSpy).toHaveBeenNthCalledWith(1, {
+                    type: '[ext] show-dialog-message',
+                    payload: {
+                        message:
+                            'The application was reloaded because of changes in a higher layer.',
+                        shouldHideIframe: false
+                    }
+                });
+                expect(reloadSpy).toHaveBeenCalled();
+                Object.defineProperty(window, 'location', {
+                    value: location
+                });
+            }, 5 * 1000);
+        });
     });
-});
+});
```

Comment:
It looks like the file is not correctly formatted either, please do so.

Reply:
Fixed in [652a18e](https://github.com/SAP/open-ux-tools/pull/2536/commits/652a18e63fd64cf9f64e2a3b65d78254075b0fbb)